MoltenVK/MoltenVK/GPUObjects/MVKQueryPool.h

```diff
@@ -35,6 +35,58 @@ class MVKCommandEncoder;
 #pragma mark -
 #pragma mark MVKQueryPool
 
+/**
+ * Index of a GPU query and associated timestamp sample (if the query is a timestamp query).
+ */
+struct ActivatedQueryInfo
+{
+	uint32_t queryIndex;
+	uint32_t timestampSampleIndex;
+};
+
+/**
+ * CPU and GPU timestamps at two points in time.
+ */
+struct TimestampCorrelationMarker {
+	MTLTimestamp cpuStart;
+	MTLTimestamp cpuEnd;
+	MTLTimestamp gpuStart;
+	MTLTimestamp gpuEnd;
+};
+
+/**
+ * Manages buffers used for resolving timestamp counter samples.
+ */
+class MVKTimestampBuffers
```

Comment:
All classes in MoltenVK subclass from `MVKBaseObject`. There is substantial functionality in that class, and its subclass hierarchy.